I've been able to test `integration-tests/infinispan-client ` successfully locally.

It looks like there is a more serious issue with `integration-tests/grpc-plain-text-gzip` as the produced binaries fail to respond correctly:

```
[ERROR] Failures:
[ERROR]   HelloWorldEndpointIT>HelloWorldEndpointTestBase.testHelloWorldServiceUsingBlockingStub:22
Expecting actual:
  "{"details":"Error id 887631a4-cfdf-4583-852e-8fa9a180fda3-1","stack":""}"
to start with:
  "Hello neo"

[ERROR]   HelloWorldEndpointIT>HelloWorldEndpointTestBase.testHelloWorldServiceUsingMutinyStub:29
Expecting actual:
  ""
to start with:
  "Hello neo-mutiny"

[ERROR] Errors:
[ERROR]   HelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingBlockingStub:47 » StatusRuntime CANCELLED: Failed to read message.
[ERROR]   HelloWorldServiceIT.testHelloWorldServiceUsingMutinyStub » StatusRuntime CANCELLED: Failed to read message.
[ERROR]   VertxHelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingBlockingStub:47 » StatusRuntime INTERNAL: No value received for unary call
[ERROR]   VertxHelloWorldServiceIT>HelloWorldServiceTestBase.testHelloWorldServiceUsingMutinyStub:55 » Timeout
[INFO]
[ERROR] Tests run: 6, Failures: 2, Errors: 4, Skipped: 0
```

I have spotted the characteristic warning where a heuristic fails to handle an `Unsafe` field:

```
Warning: RecomputeFieldValue.FieldOffset automatic substitution failed. The automatic substitution registration was attempted because a call to sun.misc.Unsafe.objectFieldOffset(Field) was detected in the static initializer of com.google.protobuf.UnsafeUtil. Detailed failure reason(s): The argument of sun.misc.Unsafe.objectFieldOffset(java.lang.reflect.Field) is not a constant value or a field load that can be constant-folded., Could not determine the field where the value produced by the call to sun.misc.Unsafe.objectFieldOffset(Field) for the field offset computation is stored. The call is not directly followed by a field store or by a sign extend node followed directly by a field store.
```

/cc @zakkak it's not related to your efforts wrt `UnsafeAccessedFieldBuildItem` (it's a 3.2 backport branch here) but that might be of interest to you as well.

--- a/extensions/grpc-common/deployment/src/main/java/io/quarkus/grpc/common/deployment/GrpcCommonProcessor.java
+++ b/extensions/grpc-common/deployment/src/main/java/io/quarkus/grpc/common/deployment/GrpcCommonProcessor.java
@@ -75,7 +75,8 @@ public class GrpcCommonProcessor {
                 .addRuntimeInitializedClass("io.grpc.netty.Utils")
                 .addRuntimeInitializedClass("io.grpc.netty.NettyServerBuilder")
                 .addRuntimeInitializedClass("io.grpc.netty.NettyChannelBuilder")
-                .addRuntimeInitializedClass("io.grpc.internal.RetriableStream");
+                .addRuntimeInitializedClass("io.grpc.internal.RetriableStream")
+                .addRuntimeReinitializedClass("com.google.protobuf.UnsafeUtil");
         return builder.build();
     }
 

--- a/extensions/grpc-common/runtime/pom.xml
+++ b/extensions/grpc-common/runtime/pom.xml
@@ -100,8 +100,27 @@
             <artifactId>quarkus-vertx</artifactId>
         </dependency>
         <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-core</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.google.code.findbugs</groupId>
+                    <artifactId>jsr305</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.codehaus.mojo</groupId>
+                    <artifactId>animal-sniffer-annotations</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>com.google.android</groupId>
+                    <artifactId>annotations</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
             <groupId>org.graalvm.nativeimage</groupId>
             <artifactId>svm</artifactId>
+            <scope>provided</scope>
         </dependency>
     </dependencies>
 

--- a/extensions/grpc-common/runtime/src/main/java/io/quarkus/grpc/common/runtime/graal/GrpcSubstitutions.java
+++ b/extensions/grpc-common/runtime/src/main/java/io/quarkus/grpc/common/runtime/graal/GrpcSubstitutions.java
@@ -2,6 +2,7 @@ package io.quarkus.grpc.common.runtime.graal;
 
 import static io.grpc.InternalServiceProviders.getCandidatesViaHardCoded;
 
+import java.lang.reflect.Field;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
@@ -11,6 +12,8 @@ import java.util.ServiceConfigurationError;
 import com.oracle.svm.core.annotate.Alias;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
+
+import sun.misc.Unsafe;
 
 @SuppressWarnings("unused")
 @TargetClass(className = "io.grpc.ServiceProviders")
@@ -77,6 +80,20 @@ interface Target_io_grpc_ServiceProviders_PriorityAccessor<T> { // NOSONAR
 
     @Alias
     int getPriority(T provider);
+}
+
+@TargetClass(className = "com.google.protobuf.UnsafeUtil")
+final class Target_com_google_protobuf_UnsafeUtil {
+    @Substitute
+    static sun.misc.Unsafe getUnsafe() {
+        try {
+            Field theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
+            theUnsafe.setAccessible(true);
+            return (Unsafe) theUnsafe.get(null);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
 }
 
 @SuppressWarnings("unused")

--- a/extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java
+++ b/extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java
@@ -205,7 +205,6 @@ class InfinispanClientProcessor {
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClass(InfinispanClientName.class).build());
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClass(Remote.class).build());
 
-        systemProperties.produce(new SystemPropertyBuildItem("io.netty.noUnsafe", "true"));
         hotDeployment
                 .produce(new HotDeploymentWatchedFileBuildItem(META_INF + File.separator + DEFAULT_HOTROD_CLIENT_PROPERTIES));
 

